### PR TITLE
Create decorator classes

### PR DESCRIPTION
**Pull request by edenlisk**
In this PR, I did the following:
1. Create a `Nameable` to raise `NotImplementedError` if the `correct_name` method is not implemented.
2. Create a `Decorator` base class
3. Create a `CapitalizeDecorator` class to capitalize the name
4. Create a `TrimmerDecorator` class to trim long name

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'decorator'
```
Comment (TedLivist) [inline capitalize_decorator.rb]: - [ ] Ruby throws an error when I try to check your program. Kindly note that `require` is often reserved for external packages like gems, while `require_relative` is used for local files.

  ```
  require_relative 'decorator'
  ```

  eieod

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'decorator'
```
Comment (TedLivist) [inline trimmer_decorator.rb]: - [ ] Kindly use `require_relative` instead

  ```
  require_relative 'decorator'
  ```

File: decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'nameable'
```
Comment (TedLivist) [inline decorator.rb]: - [ ] Kindly import your class correctly.

  ```
  require_relative 'nameable'
  ```

File: person.rb [hunk 4]
```diff
@@ -1,8 +1,10 @@
-class Person
+require 'nameable'
```
Comment (TedLivist) [inline person.rb]: - [ ] Kindly import the class correctly.

  ```
  require_relative 'nameable'
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @edenlisk,

Good job so far!
✔️ Implementation of decorator
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @edenlisk 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlight
- Good job working on the suggested changes 👍🏽

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**